pkg/kubelet/pluginmanager/reconciler/reconciler.go

```diff
@@ -100,7 +100,11 @@ func (rc *reconciler) getHandlers() map[string]cache.PluginHandler {
 	rc.RLock()
 	defer rc.RUnlock()
 
-	return rc.handlers
+	var copyHandlers = make(map[string]cache.PluginHandler)
+	for pluginType, handler := range rc.handlers {
+		copyHandlers[pluginType] = handler
+	}
+	return copyHandlers
```

Comment:
Can you explain why we need a copy here?

Reply:
A data race. You can see my second modify place `fix reconcile's field handler data race problem.`.
Here we return reference of handler and release the lock which control the race of handler.
Support that one routine is use this reference of handler. The other routine use `AddHandler ` to set it. The lock does promise this kind race.

Reply:
I don't have enough context on this part of the code to know if it is better to copy this or add proper locking around accessing it.

/cc @derekwaynecarr can hopeully find the right person to look at it.